shading/lidar/Makefile

```diff
@@ -0,0 +1,385 @@
+
+# Makefile for lidar shading, contours and cliffs
+# using folder make_data, where clipping and tiling geometries are stored
+# run (cd make_data && ./tiling.sh) before this makefile !!
+
+# # testing 
+# export input_dmr35 = input/DMR35_ZBGIS/DMR3-5_10/dmr3_5_10.tif
+# export input_dmr50 = input/DMR50_ZBGIS/data_sjtsk03
+# export input_pl = input/polsko-geoportal-dem/tif
+# export input_srtm = input/SRTM_1arc_v3
+# export make_data = make_data_test
+
+# production
+export input_root = e:\mapy
+# canot use /e/mapy, windows make wildcard returns nothing
+#   or use relative ../ - but not working in my symlink situation
+export input_dmr35 = $(input_root)/DMR3-5_ZBGIS_TIFF/DMR3-5_10/dmr3_5_10.tif
+export input_dmr50 = $(input_root)/DMR5-0_ZBGIS/data_sjtsk03
+export input_pl = $(input_root)/polsko-geoportal-dem/tif
+export input_srtm = $(input_root)/SRTM_1arc_v3
+export make_data = make_data
+
+shading50files = $(patsubst $(input_dmr50)/%.tif,build/dmr50/%/shading.tif,$(wildcard $(input_dmr50)/*.tif))
+cliffs50files = $(patsubst $(input_dmr50)/%.tif,build/dmr50/%/cliffs.gpkg,$(wildcard $(input_dmr50)/*.tif))
+contour50files = $(patsubst $(input_dmr50)/%.tif,build/dmr50/%/contours.gpkg,$(wildcard $(input_dmr50)/*.tif))
+
+pl_dem_files = $(patsubst $(make_data)/pl/%.gpkg,build/pl/retiled/%.tif,$(wildcard $(make_data)/pl/*.gpkg))
+shadingfiles_pl = $(patsubst $(make_data)/pl/%.gpkg,build/pl/%/shading.tif,$(wildcard $(make_data)/pl/*.gpkg))
+cliffsfiles_pl = $(patsubst $(make_data)/pl/%.gpkg,build/pl/%/cliffs.gpkg,$(wildcard $(make_data)/pl/*.gpkg))
+contourfiles_pl = $(patsubst $(make_data)/pl/%.gpkg,build/pl/%/contours.gpkg,$(wildcard $(make_data)/pl/*.gpkg))
+
+all: build/cliffs.gpkg build/contours.gpkg 
+# all:  build/shading_web.tif.ovr build/cliffs.gpkg build/contours.gpkg build/contours_clipped.gpkg
+
+currdir = $(shell pwd)
+export sloperamp = $(currdir)/slope.ramp
+# automatic variables
+# $@ - target  |  $^ - all prerequisites space delimited | $< - first prerequisite
+# $* - stem from implicit rule match "%" | $(@D) - directory part of $@
+# https://www.gnu.org/software/make/manual/make.html#Pattern-Rules
+# https://www.gnu.org/software/make/manual/html_node/Quick-Reference.html#Quick-Reference
+
+define make_shading
+	mkdir -p $(@D)
+	cd $(@D) && $(MAKE) -f $(currdir)/makefile-shading \
+		input=$(currdir)/$< \
+		&& $(MAKE) -f $(currdir)/makefile-shading clean
+endef
+
+
+# DMR 3.5 GKU
+
+build/dmr35/sk_dmr35_cut.tif: $(input_dmr35)
+	mkdir -p $(@D)
+	gdalwarp \
+		-of GTiff -ot Float32 -co COMPRESS=LZW -co TILED=YES -co BIGTIFF=YES \
+		-co NUM_THREADS=ALL_CPUS -wo NUM_THREADS=ALL_CPUS -multi \
+		-cutline $(make_data)/dmr35-cut.geojson -cl dmr35-cut -crop_to_cutline \

```

Comment:
cutline je potrebny? nestacia NODATA z DMR geotiffu?

Reply:
Tu konkretne som povodne len chcel zrychlit gdalwarp, aby newarpoval to, kde uz je lidar. Potom som ale navyse zistil, ze dokonca niekde v pohranici s Madarskom je DMR3.5 uplne vedla a bolo by lepsie rezat presne po hranici (nodata zacinaju az 500m za hranicou).

Reply:
Skus to s tym `Slovakia_JTSK03_to_JTSK.gsb`, ide aj kus za hranice:

![image](https://user-images.githubusercontent.com/636095/104136980-d561f300-5399-11eb-896e-117fcfeebd54.png)

Reply:
Nemyslel som posunutie. Na hranici s Madarskom presne za hranicou chybaju vyskove udaje, aj ked nodata tam este nie je. Hore DMR3.5, dole SRTM:
![image](https://user-images.githubusercontent.com/36862415/104137454-19a2c280-539d-11eb-90ba-d2dccdb5a5c5.png)